source/Calamari.Common/CalamariFlavourProgram.cs

```diff
@@ -25,21 +25,21 @@
 
 namespace Calamari.Common
 {
-    public abstract class CalamariFlavourProgram
+    public abstract class  CalamariFlavourProgram
     {
         protected readonly ILog Log;
 
         protected CalamariFlavourProgram(ILog log)
         {
-            Log = log;
+            Log = new RedactedValuesLogger(log);
```

Comment:
Will using this everywhere have any performance/memory implications? What about if it's used to redact large log strings? If so, should there be XML comments warning of that?  
(not saying there definitely are, but just a reminder to consider these things)

Reply:
I think there will be a minor hit but because the vast majority of times we don't have any values to redact, it will just be returning the value without attempting any sort of replacements anyway. If there are values to redact then we definitely want to pay the price for replacing those strings.

The main problem I'm trying to solve with this change is that if we move things around (that's what I'm planning to do in subsequent PRs) we can't rely on the fact that the logger is a redacted logger, and if it's not, we can't rely on things getting redacted correctly, unless we say that the default logger is a redacted logger.

I ran into this issue during my spike work and I wanted to make sure we covered it before any secrets were leaked into logs.